napari_arboretum/visualisation/base_plotter.py

```diff
@@ -0,0 +1,135 @@
+import abc
+from typing import Sequence
+
+import napari
+import numpy as np
+from qtpy.QtWidgets import QWidget
+
+from ..graph import build_subgraph, layout_subgraph
+from ..tree import ColorType
+
+GUI_MAXIMUM_WIDTH = 600
+
+__all__ = ["TreePlotterBase", "TreePlotterQWidgetBase"]
+
+
+class TreePlotterBase(abc.ABC):
+    """
+    Base class for a `napari.layers.Tracks` plotter.
+
+    This class is designed to handle the translation from a `napari.layers.Tracks`
+    layer to visual objects that can be plotted (e.g. lines, text). As such its
+    only state is the ``_tracks`` attribute.
+
+    This is not designed to actually render the plotting objects objects.
+    Sub-classes should do that by impelmenting the abstract methods defined below.
+    """
+
+    @property
+    def tracks(self) -> napari.layers.Tracks:
+        """
+        The napari tracks layer associated with this plotter.
+        """
+        if not hasattr(self, "_tracks"):
+            raise AttributeError("No tracks set on this plotter.")
+        return self._tracks
+
+    @tracks.setter
+    def tracks(self, track_layer: napari.layers.Tracks):
+        self._tracks = track_layer
+
+    def draw_tree(self, track_id: int) -> None:
+        """
+        Plot the tree containing ``track_id``.
+        """
+        root, subgraph_nodes = build_subgraph(self.tracks, track_id)
+        self.edges, self.annotations = layout_subgraph(root, subgraph_nodes)
+
+        self.set_title(f"Lineage tree: {track_id}")
+
+        self.plot_branches()
+
+        # labels
+        for a in self.annotations:
+            # change the alpha value according to whether this is the selected
+            # cell or another part of the tree
+            color = a.color
+            color[3] = 1 if a.label == str(track_id) else 0.25
```

Comment:
I think the comment above covers clearly what's going on here?

Reply:
IMO a variable beats a comment https://blog.webdevsimplified.com/2020-02/magic-numbers/ here for example. Here, for example, it could be `opaque=1` and `more_transparent=0.25`